packages/next/client/components/app-router.client.tsx

```diff
@@ -353,13 +354,15 @@ export default function AppRouter({
                 url: canonicalUrl,
               }}
             >
-              {HotReloader ? (
-                <HotReloader assetPrefix={assetPrefix}>
-                  {cache.subTreeData}
-                </HotReloader>
-              ) : (
-                cache.subTreeData
-              )}
+              <ErrorBoundary errorComponent={GlobalErrorComponent}>
```

Comment:
This boundary should wrap around the AppRouter component instead.

Changing the default export to something like:

```
export default function AppRouter(props) {
  return <ErrorBoundary errorComponent={GlobalErrorComponent}>
     <Router {...props} />
  </ErrorBoundary>
}
```

The existing function would be renamed to `Router` in this.